recipes/llvm/all/conanfile.py

```diff
@@ -48,7 +50,11 @@ def source(self):
         os.rename(extracted_dir, self._source_subfolder)
 
     def configure(self):
-        tools.check_min_cppstd(self, '14')
+        if self.settings.compiler.get_safe("cppstd"):
+            tools.check_min_cppstd(self, '14')
+
+        if self.settings.compiler == "Visual Studio" and Version(self.settings.compiler.version) < "19.1":
+            raise ConanInvalidConfiguration("Need MSVC >= 19.1")
```

Comment:
I think you're too restrictive. Visual Studio 2017 is version 15.
(see the generator section of `cmake --help`)

Reply:
Nope, see the previous build error

Reply:
19.0.24215.1 is the version of the C compiler, not the version of visual studio.

Look at personal your conan settings file: `$HOME/.conan/settings.yml`.
There exists no Visual Studio compiler whose version >= 19.1.
Or check https://en.wikipedia.org/wiki/Microsoft_Visual_Studio#2019.
Visual Studio 2019 => version 16.

Reply:
https://github.com/llvm/llvm-project/blob/master/llvm/cmake/modules/CheckCompilerVersion.cmake#L53-L64

So host Visual Studio version 16.4 and lower is known to miscompile part of LLVM.

- That means only up-to-date versions of visual studio 2019 can build llvm.
- This also means that the test in `configure` must compare against the `major.minor` version of MSVC (16.4) instead of only `major` (16).